src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -234,7 +185,7 @@ private function createAutowiredDefinition(\ReflectionClass $typeHint, $id)
         $argumentDefinition = $this->container->register($argumentId, $typeHint->name);
         $argumentDefinition->setPublic(false);
 
-        $this->populateAvailableType($argumentId, $argumentDefinition);
+        $this->populateAvailableTypes();
         $this->completeDefinition($argumentId, $argumentDefinition);
```

Comment:
Yeah it is, this function creates a new definition which is not included in the types map yet.
So this call add it into the map.